### PR TITLE
Attempts to fix build by updating to latest webpack things

### PR DESCRIPTION
**Pull request by adriancole**
The build now passes, but the css looks off in the UI. Will need help
completing this.

See #29

### CODE REVIEW

**Comment by adriancole** [comment]
@charlescui @abesto @rogeralsing (anyone) halp! I spent a couple hours on this and the build works, but the UI looks like the css is not right..

screen shot 2017-03-21 at 9 48 41 pm

File: config/webpack.prod.js [hunk 1]
```diff
@@ -16,13 +16,8 @@ module.exports = webpackMerge(commonConfig, {
         chunkFilename: '[id].[hash].chunk.js'
     },
 
-    htmlLoader: {
-        minimize: false // workaround for ng2
-    },
-
     plugins: [
-        new webpack.NoErrorsPlugin(),
-        new webpack.optimize.DedupePlugin(),
```
Comment (adriancole) [inline config/webpack.prod.js]: the omissions and renames here were made to correct errors in the log

**Comment by apottere** [comment]
Are you looking for something like this?

image

I downgraded bootstrap to 3.3.7 and changed the header html a bit:
```
<nav class="navbar navbar-fixed-top navbar-inverse">
    <div class="container">
        <a class="navbar-brand" href="#">Zipkin <small>- Investigate system behavior</small></a>
        <ul class="nav navbar-nav">
            <li class="nav-item">
                <a class="nav-link" href="#">Traces</a>
            </li>
            <li class="nav-item">
                <a class="nav-link" href="/#/dependencies">Dependencies</a>
            </li>
        </ul>
    </div>
</nav>
```

**Comment by apottere** [comment]
Ah, I just checked the diff and you updated from alpha 5 to alpha 6, so this probably isn't what you want.  I'll keep digging.

**Comment by apottere** [comment]
You can get something like this with the current bootstrap:

image

Header:
```
<nav class="navbar navbar-toggleable-sm fixed-top navbar-inverse bg-inverse">
    <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
        <span class="navbar-toggler-icon"></span>
    </button>
    <a class="navbar-brand" href="#">Zipkin <small>- Investigate system behavior</small></a>

    <div class="collapse navbar-collapse" id="navbarSupportedContent">
        <ul class="navbar-nav mr-auto">
            <li class="nav-item">
                <a class="nav-link" href="#">Traces</a>
            </li>
            <li class="nav-item">
                <a class="nav-link" href="/#/dependencies">Dependencies</a>
            </li>
        </ul>
    </div>
</nav>
```

**Comment by apottere** [comment]
The latter code was just pulled from the bootstrap docs with the current nav items plugged in.

**Comment by apottere** [comment]
I can't get the master branch running to see what it was like before, but it seems unlikely that a webpack update could have caused the issues with the nav.

**Comment by adriancole** [comment]
thanks @apottere  pushed a commit to fix the navbar.

The detail pane seems off.. here's an old screenshot
image

Here's the result of the below trace

screen shot 2017-03-22 at 10 44 06 am

```bash
$ curl -s localhost:9411/api/v1/spans -H'Content-Type:application/json' -d '[
    {
      "traceId": "2e0456da58cc75f9",
      "id": "2e0456da58cc75f9",
      "name": "get",
      "timestamp": 1490150337461000,
      "duration": 1429325,
      "annotations": [
        {
          "timestamp": 1490150337461000,
          "value": "sr",
          "endpoint": {
            "serviceName": "brave-webmvc-example",
            "ipv4": "192.168.99.1"
          }
        },
        {
          "timestamp": 1490150338890325,
          "value": "ss",
          "endpoint": {
            "serviceName": "brave-webmvc-example",
            "ipv4": "192.168.99.1"
          }
        }
      ],
      "binaryAnnotations": [
        {
          "key": "http.status_code",
          "value": "200",
          "endpoint": {
            "serviceName": "brave-webmvc-example",
            "ipv4": "192.168.99.1"
          }
        },
        {
          "key": "http.url",
          "value": "/a",
          "endpoint": {
            "serviceName": "brave-webmvc-example",
            "ipv4": "192.168.99.1"
          }
        }
      ]
    },
    {
      "traceId": "2e0456da58cc75f9",
      "id": "4b722eee46e4746a",
      "name": "get",
      "parentId": "2e0456da58cc75f9",
      "timestamp": 1490150338031000,
      "duration": 845960,
      "annotations": [
        {
          "timestamp": 1490150338031000,
          "value": "cs",
          "endpoint": {
            "serviceName": "brave-webmvc-example",
            "ipv4": "192.168.99.1"
          }
        },
        {
          "timestamp": 1490150338034000,
          "value": "sr",
          "endpoint": {
            "serviceName": "brave-webmvc-example",
            "ipv4": "192.168.99.1"
          }
        },
        {
          "timestamp": 1490150338876000,
          "value": "ss",
          "endpoint": {
            "serviceName": "brave-webmvc-example",
            "ipv4": "192.168.99.1"
          }
        },
        {
          "timestamp": 1490150338876960,
          "value": "cr",
          "endpoint": {
            "serviceName": "brave-webmvc-example",
            "ipv4": "192.168.99.1"
          }
        }
      ],
      "binaryAnnotations": [
        {
          "key": "http.status_code",
          "value": "200",
          "endpoint": {
            "serviceName": "brave-webmvc-example",
            "ipv4": "192.168.99.1"
          }
        },
        {
          "key": "http.url",
          "value": "http://localhost:8081/b",
          "endpoint": {
            "serviceName": "brave-webmvc-example",
            "ipv4": "192.168.99.1"
          }
        },
        {
          "key": "http.url",
          "value": "/b",
          "endpoint": {
            "serviceName": "brave-webmvc-example",
            "ipv4": "192.168.99.1"
          }
        }
      ]
    }
  ]'
```

**Comment by apottere** [comment]
@adriancole so I found this when poking around:
>Bootstrap 4 introduced the .bg-* class, as well as the .navbar-light and .navbar-dark classes.
http://www.quackit.com/bootstrap/bootstrap_4/tutorial/bootstrap_navbars.cfm

AFAICT, the `.navbar-dark` isn't supplied by the version of bootstrap that's being pulled in by NPM.  I'll keep digging, but it's possible we're getting a bad version of the bootstrap alpha.

**Comment by apottere** [comment]
Ah, navbar-dark was renamed to navbar-inverse: https://github.com/twbs/bootstrap/commit/80674b0c58e3fd8ce76ba94d1075d62c2db6af37#diff-df1f74faf207a474f1d78c86393221c8

**Pull request closed without merging**